src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ToolsApiScannerInstaller.java

```diff
@@ -220,4 +231,19 @@ private void removeOldVersionFileIfExists(File scannerExpansionDirectory) {
         }
     }
 
+
+    private Certificate connectAndGetServerCertificate(HttpUrl httpsServer) {
+        HttpsURLConnection httpsConnection = null;
+        try {
+            httpsConnection = (HttpsURLConnection) httpsServer.url().openConnection();
+            httpsConnection.connect();
+            Certificate[] certificates = httpsConnection.getServerCertificates();
+            httpsConnection.disconnect();
+
+            return certificates[0];
+        } catch (IOException e) {
+            logger.errorAndDebug("Could not get Black Duck server certificate which is required for managing the local keystore - communicating to the server will have to be configured manually: " + e.getMessage(), e);
+            return null;
+        }
+    }
```

Comment:
Connecting to the server to just fetch the certificate was done in the past: https://github.com/blackducksoftware/blackduck-common/pull/307/files 

Ideally using the connection we already establish for downloading the CLI would be ideal but looking at the DefaultBlackDuckHttpClient, I didn't see a clean way to do this.

Reply:
Could we maybe have Chris review this error message? If this fails it might be better to try the download again. It would be pretty difficult to configure manually?

Reply:
When I first saw this I thought "oh? that's an option? how would one do that?" ... Not sure if it's mentioned in docs somewhere. 

@sig-cpotts Thoughts?   


This wording is a slight variation on what is logged today for essentially the same error through the "original" flow/API: https://github.com/blackducksoftware/blackduck-common/blob/84f9db2dec10892841e243c9db4fdfab7514c72b/src/main/java/com/synopsys/integration/blackduck/keystore/KeyStoreHelper.java#L45

Reply:
"How would one configure that manually" is for sure a valid question a user may have. : )
Would be nice to indicate in some brief way what that means, i.e. manually add server cert to JRE instance keystore. (Though there is a limit to how much detail can be in an error message.)
If there is potential that downloading again might resolve the issue, we should definitely indicate that to help the user save time/effort.
(We have a somewhat related note in the "Common Problems" section of the doc: https://sig-product-docs.synopsys.com/bundle/integrations-detect/page/troubleshooting/solutions.html#ariaid-title17)